ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -223,23 +223,28 @@ private[commands] class CommandTracker[Context](maxDeduplicationTime: () => JDur
                 s"A command with id $commandId is already being tracked. CommandIds submitted to the CommandTracker must be unique."
               ) with NoStackTrace
             }
-            val commandTimeout = {
-              lazy val maxDedup = maxDeduplicationTime()
-              val dedup = commands.deduplicationTime.getOrElse(
-                ProtoDuration.of(maxDedup.getSeconds, maxDedup.getNano)
-              )
-              Instant.now().plusSeconds(dedup.seconds).plusNanos(dedup.nanos.toLong)
+            maxDeduplicationTime() match {
+              case None =>
+                emit(
+                  resultOut,
+                  submitRequest.map { _ =>
+                    val status = GrpcStatus.toProto(ErrorFactories.missingLedgerConfig().getStatus)
+                    Left(NotOkResponse(commandId, status))
+                  },
+                )
```

Comment:
Why would we delay the check until here? Wouldn't it make more sense to fail fast during startup as we know if the config is missing or not.

Reply:
Well, the config can be missing for the first 5 seconds, and then show up.

`LedgerConfigSubscription` is mutable; the state can change.

Reply:
Can't we validate it though before we get to this point? We're creating the tracker in an invalid state then from the beggining and this config validation seem to leak way too deep into the tracker.